RIOT-OS/RIOT
# tests/candev: add mcp2515 + driver/mcp2515: add driver

### Contribution description
* Add the mcp2515 driver code from #6276
* Apply some minor fixes so it works with the current RIOT API (Because the driver was written > 4 years ago, and RIOT has obviously changed a bit since then) (Also ran uncrustify while I was at it)
* Add mcp2515 support to the candev app (see: #13360)

### Testing procedure
* Adapt pinconfiguration in Makefile to match your own hardware (e.g. interrupt pin)
* Connect the board over USB with a linux laptop through means of a PCAN-USB cable (or something similar). Alternatively you can connect two boards to each other and receive the output of one on the other. This would remove the need for an expensive PCAN cable, but adds some complexity. 
* Make sure libsocketcan and canutils are installed and running (see: https://github.com/RIOT-OS/RIOT/tree/master/tests/conn_can)
* Start candump on can0
* In the test-app shell: the `send` command will send a CAN message to can0. It should become visible in candump. Optionally you can pass some bytes (in decimal notation) as arguments (e.g. `send 12 34 56`). If no arguments are supplied, three default bytes will be sent. 
* To test the receive part, we can send data from Linux with cansend (e.g. `cansend can0 001#AABB`) or with cangen (e.g. `cangen can0 -v -n 1`).
* In the test-app, the received bytes will be stored in a FIFO buffer and can be retrieved by calling `receive`. (an optional argument n can be passed to receive multiple can-messages). If there are no CAN-messages in the buffer when `receive` is called, it will block until a message is received.



### Issues/PRs references
replaces #6276
